_posts/2022-05-23-fine-grained-access-control-for-multitenancy.markdown

```diff
@@ -0,0 +1,326 @@
+---
+layout: post
+title: Fine Grained Access Control for Multi-tenancy
+description: A how to guide for architecting multitenant backend applications
+date: 2022-05-23
+author: Leon Ticharwa
+image: img/blog/multitenancy/multitenancy.png
+tags: ['multitenant', 'typescript', 'tutorial', 'dynamodb', 'lambda', 'api gateway', 'authorization', 'authentication', 'apiGateway', 'jwt', 'json web token', 'cognito']
+---
+
+## Introduction
+
+Architecting multi-tenant cloud native software can be quite challenging due to the broad scope of technical know-how required to execute such an undertaking.
+What's even more challenging is finding cohesive guides/tutorials that take a top down approach to the somewhat elusive topic of multitenancy. This article is an attempt to produce a distilled "how-to" guide for those who may find themselves bogged down while building multi-tenant applications. You clone the repo containing the reference code from here [reference code here](https://github.com/MechanicalRock/Multitenancy-AuthorizationAuthentication).
```

Comment:
Minor; consistency with multi-tenancy. Sometimes it's multi-tenancy, other times it's multitenancy (no dash).